Process_of_Authoring_GitHub_Actions_for_Azure.md

```diff
@@ -0,0 +1,98 @@
+
+**Note:** It is recommended that team handles are used for maintaining permissions instead of adding individual member's github handles.
+
+- Kindly request the corresponding Azure product teams to include their team handle as "Admins"/"Maintainers" to facilitate raising any commits from their side for feature enhancements or for fixing supportability issues
+
+
+## ReadMe.md &amp; action.yml files
+
+Once the Action development is complete, ensure that it is documented well in the **readme.md** file at the root of the repo of the Master branch. Please note that this content is used when the Action is published to the GitHub Marketplace. Hence including a sample workflow which covers the best usage of the Action is always recommended.
+
+Also ensure that the **action.yml** is present in the root of the repo and has the Action name &amp; description well defined. Also choose the icon &amp; color suitably under branding as these are used when the Action is listed in Marketplace catalog.
+
+Note: All actions published from **Azure** GitHub org are branded using the Azure icon irrespective of the icon used in the action.yml. This is to leverage the recall value of Azure brand and icon as compared to the individual icons of various Azure services. 
+
+## Action Versioning
+
+**Versioning Guidance for GH actions that should be adhered to before a new action or an enhancement to an existing action is released:** 
+
+Official Release Management Guidance from GH platform team: https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/about-actions#using-release-management-for-actions
+
+**Note:** Introduce a new major version of the branch (releases/v2) and tag (v2) if changes break the existing workflows using the previous version. In case of Minor version releases, ensure to move the major version tag to always point to the most recent minor version tag to ensure all the existing users of the major version would automatically gain the benefits from the new enhancements added in the minor version release.
+
+
+## Publish the action to Marketplace
+
+### Pre-requisites:
+
+- Ensure that the Legal OSS approval process for the repo is complete. Search for the status of the work item created with your repo name here: https://ossmsft.visualstudio.com/Reviews/\_workitems/ .
+
+&quot;DO NOT PUBLISH THE ACTION TO THE MARKETPLACE UNTIL LEGAL REVIEW IS COMPLETE&quot;
+
+- Versioning guidelines mentioned above section are adhered to and you have a version tag
+- Well documented readme.md is in place at the root of the repo
+- V-team is involved for review process and sign off by mailing @[ACE team Alias](mailto:ace-team@github.com)
+
+Finally [publish the action to GitHub Marketplace](https://docs.github.com/free-pro-team@latest/actions/creating-actions/publishing-actions-in-github-marketplace)
```

Comment:
Should we have a approval based workflow to publish to market place. This will bring traceability and also have right check and balance.

Reply:
Sure Muthu.. Should we replace this guidance once we have such automation in place?

Reply:
Yes. We can track a feature to add that workflow automation.

Reply:
https://github.com/github/GitHub-EcoSystem/issues/1104